common/osqp_interface/test/test_osqp_interface.cpp

```diff
@@ -23,67 +23,109 @@
 namespace
 {
 using autoware::common::osqp::float64_t;
-/// Problem taken from https://github.com/osqp/osqp/blob/master/tests/basic_qp/generate_problem.py
+// Problem taken from https://github.com/osqp/osqp/blob/master/tests/basic_qp/generate_problem.py
+//
+// min  1/2 * x' * P * x  + q' * x
+// s.t. lb <= A * x <= ub
+//
+// P = [4, 1], q = [1], A = [1, 1], lb = [   1], ub = [1.0]
+//     [1, 2]      [1]      [1, 0]       [   0]       [0.7]
+//                          [0, 1]       [   0]       [0.7]
+//                          [0, 1]       [-inf]       [inf]
+//
+// The optimal solution is
+// x = [0.3, 0.7]'
+// y = [-2.9, 0.0, 0.2, 0.0]`
+// obj = 1.88
+
 // cppcheck-suppress syntaxError
 TEST(TestOsqpInterface, BasicQp) {
+  using autoware::common::osqp::CSC_Matrix;
+  using autoware::common::osqp::calCSCMatrix;
+  using autoware::common::osqp::calCSCMatrixTrapezoidal;
+
   auto check_result =
-    [](const std::tuple<std::vector<float64_t>, std::vector<float64_t>, int, int> & result) {
-      EXPECT_EQ(std::get<2>(result), 1);
-      EXPECT_EQ(std::get<3>(result), 1);
-      ASSERT_EQ(std::get<0>(result).size(), size_t(2));
-      ASSERT_EQ(std::get<1>(result).size(), size_t(2));
-      EXPECT_DOUBLE_EQ(std::get<0>(result)[0], 0.3);
-      EXPECT_DOUBLE_EQ(std::get<0>(result)[1], 0.7);
-      EXPECT_DOUBLE_EQ(std::get<1>(result)[0], -2.9);
-      EXPECT_NEAR(std::get<1>(result)[1], 0.2, 1e-6);
```

Comment:
![image](https://user-images.githubusercontent.com/20228327/153786985-38b48add-b5df-4fd3-a012-074acf871b1e.png)

Reply:
![image](https://user-images.githubusercontent.com/20228327/153787016-449a4b61-bc60-4448-b3e6-67fd34ba22a0.png)
https://github.com/autowarefoundation/autoware.universe/blob/tier4/proposal/common/osqp_interface/src/osqp_interface.cpp/#L243
https://github.com/tier4/autoware.universe/tree/feature/osqp-add-functions

Reply:
@TakaHoribe Thank you for the fix! I confirmed the code and the test passed. :pray: 
@rej55 It seems that implementation to extract lagrange multiplier was wrong.